include/zephyr/bluetooth/gatt.h

```diff
@@ -1463,10 +1463,10 @@ struct bt_gatt_discover_params {
 	uint16_t end_handle;
 	/** Discover type */
 	uint8_t type;
-#if defined(CONFIG_BT_GATT_AUTO_DISCOVER_CCC)
+#if defined(CONFIG_BT_GATT_AUTO_DISCOVER_CCC) || defined(__DOXYGEN__)
 	/** Only for stack-internal use, used for automatic discovery. */
 	struct bt_gatt_subscribe_params *sub_params;
```

Comment:
Note: the comment reads as if the field is meant to be internal anyway, so it's not clear to me whether it should effectively show up in the **API** documentation?

Reply:
You're right this is a field used internally but the structure is available for the user. So, because of that, it's important that the user know that this filed is not supposed to be used.

Reply:
Mmmh then I'm confused. The field already had a comment mentioning it's internal, and in addition it was also not showing up in the doc anyway, so making it visible arguably makes things less clear (and is not aligned with how internal stuff is typically handled). The proper way to handle this would probably be to wrap the internal field in a `@cond INTERNAL_HIDDEN` / `@endcond` block

poke @gmarull

Reply:
I'm not sure we have a policy for internal struct members. I'm fine either way, as soon as it's clearly marked as internal. Internal fields should likely have a naming convention as well, e.g. `_name`.